r/R/dplyr-functions.R

```diff
@@ -255,6 +255,10 @@ nse_funcs$is_logical <- function(x, n = NULL) {
   assert_that(is.null(n))
   nse_funcs$is.logical(x)
 }
+nse_funcs$is_timestamp <- function(x, n = NULL) {
+  assert_that(is.null(n))
+  inherits(x, "POSIXt") || (inherits(x, "Expression") && x$type_id() %in% Type[c("TIMESTAMP")])
+}
```

Comment:
The other functions in this section mimic ones from `rlang`, but `is_timestamp()` doesn't exist in `rlang`.  Does this matter? Probably not?

Reply:
Yeah, that is a bit odd. There's also not a base `is.timestamp()` ASAIK (one could implement one there too, but I just needed this one).

Though {lubridate} does have them:
```
lubridate::is.Date
lubridate::is.POSIXct
lubridate::is.POSIXlt
lubridate::is.POSIXt
lubridate::is.difftime
lubridate::is.duration
lubridate::is.instant
lubridate::is.interval
lubridate::is.period
lubridate::is.timepoint
lubridate::is.timespan
```

If we want to add these + add tests for this we can (though for now I'm going to make a ticket for that: https://issues.apache.org/jira/browse/ARROW-14227)